src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -852,17 +852,21 @@ def add_permission(client, identifier, api, api_permissions):
     #         id            <- api_permissions
     #         type
 
-    application = show_application(client, identifier)
-
     resource_access_list = []
     for e in api_permissions:
-        access_id, access_type = e.split('=')
+        try:
+            access_id, access_type = e.split('=')
+        except ValueError as ex:
+            from azure.cli.core.azclierror import ArgumentUsageError
+            raise ArgumentUsageError('Usage error: Permission type should be provided, such as '
```

Comment:
The error msg could be more general because customer may specify `--api-permissions Scope` or `--api-permissions e1fe6dd8-ba31-4d61-89e7-88639da4683d Scope`

Reply:
Nice catch! Changed as requested.